tests/phpunit/E2E/Extern/SoapTest.php

```diff
@@ -61,24 +75,36 @@ public function testAuthenticationBadPassword() {
    * @expectedException SoapFault
    */
   public function testAuthenticationBadKey() {
-    $client = $this->createClient();
-    $key = $client->authenticate($this->adminUser, $this->adminPass);
-    $client->get_contact(mt_rand(), []);
+    foreach ($this->getSoapEndPoints() as $key => $url) {
+      $this->url = $url;
+      if (empty($this->url)) {
+        $this->markTestSkipped('Failed to set url for endpoint ' . $key . ' . Have you configured `cv` for testing?');
+      }
+      $client = $this->createClient();
+      $key = $client->authenticate($this->adminUser, $this->adminPass);
+      $client->get_contact(mt_rand(), []);
+    }
   }
 
   /**
    * A basic test for one SOAP function.
    */
   public function testGetContact() {
-    $client = $this->createClient();
-    $key = $client->authenticate($this->adminUser, $this->adminPass);
-    $contacts = $client->get_contact($key, array(
-      'contact_id' => 101,
-      'return.display_name' => 1,
-    ));
-    $this->assertEquals($contacts['is_error'], 0);
-    $this->assertEquals($contacts['count'], 1);
-    $this->assertEquals($contacts['values'][101]['contact_id'], 101);
+    foreach ($this->getSoapEndPoints() as $key => $url) {
+      $this->url = $url;
+      if (empty($this->url)) {
+        $this->markTestSkipped('Failed to set url for endpoint ' . $key . ' . Have you configured `cv` for testing?');
+      }
+      $client = $this->createClient();
+      $key = $client->authenticate($this->adminUser, $this->adminPass, !($key === 'traditional'));
+      $contacts = $client->get_contact($key, array(
+        'contact_id' => 101,
+        'return.display_name' => 1,
+      ), 'Failure on end point ' . $key);
+      $this->assertEquals($contacts['is_error'], 0);
```

Comment:
I'm interested in this because it partly coincides with the SSL stuff I'm doing, but I think the problem is that when you use the civicrm/soap endpoint, and so are calling authenticate with loadBootstrap TRUE in this test `(!=='traditional')`, drupal has already been bootstrapped before civi calls bootstrap. `civicrm/soap` is just drupal-speak for index.php?q=civicrm/soap, and index.php calls bootstrap. In phpunit this would be a fatal error when it hits the civi drupal UF class that does bootstrap, which maybe then gets caught somewhere and shows up here as a blank key.

You can see this in action if you run locally with my [loud notices](https://lab.civicrm.org/extensions/loudnotices) extension and then just try a [basic soap cli test](https://www.civismtp.com/drupal/accountsetup#invalidproxy) to civicrm/soap with the loadbootstrap parameter to authenticate true, and then look in ConfigAndLog. You can see drupal is already bootstrapped.

Unrelated, but you have 3 parameters to get_contact here and it only takes 2 - I think you were just trying to add some debug output.